pkg/kv/kvserver/replica_test.go

```diff
@@ -9936,7 +9935,9 @@ func TestReplicaServersideRefreshes(t *testing.T) {
 		// Regression test for #31870.
 		snap := tc.engine.NewSnapshot()
 		defer snap.Close()
-		res, err := tc.repl.sha512(context.Background(), *tc.repl.Desc(), tc.engine, nil /* diff */, roachpb.ChecksumMode_CHECK_FULL, limit.NewLimiter(rate.Inf))
+		res, err := tc.repl.sha512(context.Background(), *tc.repl.Desc(), tc.engine,
+			nil /* diff */, roachpb.ChecksumMode_CHECK_FULL,
+			quotapool.NewRateLimiter("ConsistencyQueue", quotapool.Limit(math.MaxFloat64), math.MaxInt64))
```

Comment:
NewRateLimiter is quiet on whether MaxFloat64 burst works. Good opportunity to check and update the comment there accordingly.

Reply:
I have trouble seeing how this can work btw. The burst is also the capacity of the bucket. An infinitely sized bucket limits.. nothing?

Reply:
Oh nevermind, this is the plan here. Still good to check the semantics. Sorry about noise

Reply:
not sure I follow, which comment? I deleted the custom rate limiter and based on my reading of the code there are no restrictions on the burst value in quotapool.RateLimiter

Reply:
APIs that take a float often handle the extremal values poorly, so it's good to point out explicitly when they do (at least according to my preference). It's fine if you don't want to reach into the quotapool package in this PR, though.